src/web/mjs/connectors/CxC.mjs

```diff
@@ -0,0 +1,254 @@
+        try {
+            chapterData = await this.fetchJSON(request);
+        } catch (error) {
+            if (!this.auth.accessToken) {
+                throw Error("Could not get chapters, may be age-restricted (18禁). Add an account to view chapters.");
+            } else {
+                throw Error("Could not get chapters, your account may not be configured to view age-restricted works (18禁).");
+            }
+        }
+        let chapters = [];
+        chapterData.data.data.map((chapter, index) => {
+            const chapterNum = index + 1;
+            chapters.push({
+                id: `${manga.id}/chapter/${chapter.id}`,
+                title: `#${chapterNum.toString()} - ${chapter.name}`
+            });
+        });
+        return chapters;
+    }
+
+    async _getPages(chapter) {
+        await this._login();
+        let request = new Request(`${this.api}${chapter.id}`, this.requestOptions);
+        let contentData;
+        try {
+            contentData = await this.fetchJSON(request);
+        } catch (error) {
+            throw Error("Could not get pages, chapter may require a fan subscription (粉絲免費).");
+        }
+        let imageData = [];
+        contentData.data.content.map((page) => {
+            imageData.push({
+                url: page.url,
+                imageKey: page.key
+            });
+        });
+        return imageData.map(image => this.createConnectorURI(image));
+    }
+
+    async _handleConnectorURI(payload) {
+        let response = await fetch(payload.url);
+        let encryptedImage = await response.arrayBuffer();
+        // from app.*.js 'getImgEncrypted'
+        let encTransformImage = btoa(
+            [].reduce.call(
+                new Uint8Array(encryptedImage),
+                function (e, _) {
+                    return e + String.fromCharCode(_);
+                },
+                ''
+            )
+        );
+        const token = this.auth.accessToken ? this.auth.accessToken : 'freeforcxc2021reading'; //default free content token, may change in the future
+        const tokenHash = CryptoJS.SHA512(token).toString();
+        const tokenCombo = {
+            key: tokenHash.substr(0, 64),
+            iv: tokenHash.substr(30, 32)
+        };
+        let imageComboString = this._decrypt(payload.imageKey, tokenCombo).split(':');
+        const imageCombo = {
+            key: imageComboString[0],
+            iv: imageComboString[1]
+        };
+        let decryptedImageResponse = this._decrypt(encTransformImage, imageCombo).split(",");
+        // from https://cxc.today/worker/base64ToBlob.js
+        const e = decryptedImageResponse[0].indexOf("base64") >= 0 ? atob(decryptedImageResponse[1]) : decodeURI(decryptedImageResponse[1]);
+        let n = new Uint8Array(e.length);
+        for (let i = 0; i < e.length; i++) {
+            n[i] = e.charCodeAt(i);
+        }
+        return {
+            mimeType: `image/${payload.url.split('.').pop()}`,
```

Comment:
what if the image ends with jpg and not jpeg? 

there is a method to detect mime type. something like 

```
      data = await this._blobToBuffer(data);
        this._applyRealMime(data);
        return data;
```